### PR TITLE
Feature/near flat scss

### PR DESCRIPTION
**Pull request by patrickbenjamin1**
## What's new?

- CSS flattened where possible (still some nested selectors for places where it's necessary)
- Split animations into `animated.scss` theme

## Ticket number(s) in JIRA (if internal)

ARM-XX

[board](https://rocketmakers.atlassian.net/jira/software/projects/HM/boards/172)

## Checklist

- [n/a] are your changes in Storybook?
- [n/a] are any breaking changes documented in `docs/migrating_from_oldstrong.md`?
- [n/a] does _everything_ have jsdoc?
- [n/a] is everything exported from index.ts?

### CODE REVIEW

File: module/src/components/button/button.animated.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.arm-button {
+  transition: opacity $transition-quick, background-color $transition-quick, transform $transition-quick;
+
+  &:active {
+    transform: translateY(2px);
```
Comment (adamiprinciples) [inline module/src/components/button/button.animated.scss:5]: This replicates the bug in the old armstrong in safari where if you click the top of the button, the click fails to register as the mouseup event doesn't get called.

File: module/src/components/rangeInput/rangeInput.basic.scss [hunk 2]
```diff
@@ -5,76 +5,75 @@ $range-input-track-height: 6px;
...
-      .arm-icon {
-        @include absolute-centre-self;
-      }
-    }
+  &::after {
+    @include init-pseudo;
+    position: absolute;
+    left: 0;
+    top: 0;
+    width: var(--arm-range-input-percent);
```
Comment (adamiprinciples) [inline module/src/components/rangeInput/rangeInput.basic.scss:62]: I thought we weren't using CSS variables?

Comment (jlews) [reply module/src/components/rangeInput/rangeInput.basic.scss:62]: My understanding is that we're not creating them for our general use variables, but we are using them when you need to set a property directly through JSX and the only alternative would be black-boxing it inside the JS with:
```
style={{ width: `${percent}%` }}
```

File: module/src/components/button/button.animated.scss (hunk 1 above)
Comment (patrickbenjamin1) [reply module/src/components/button/button.animated.scss:5]: haha good old safari

Comment (patrickbenjamin1) [reply module/src/components/button/button.animated.scss:5]: I'll get rid of it and let it be up to the consuming app how the button animates

**Pull request merged**